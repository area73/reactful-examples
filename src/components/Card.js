import React from 'react';


class Card extends React.Component {


  render() {
    const data = this.props;
    return (
      <div className="card">
        <img className="card_thumb" src={data.avatar_url}/>
        <div className="card__info">
          <div className="card__substract"><a href="">X</a></div>
          <div className="card__name">{data.name}</div>
          <div className="card__company">{data.company}</div>
        </div>

      </div>
    );
  }
}

export default Card;
