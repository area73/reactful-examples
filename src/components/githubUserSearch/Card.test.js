import React from 'react';
import Card  from './Card.js';

import renderer from 'react-test-renderer';


const props = {
  'login': 'area73',
  'id': 2030605,
  'node_id': 'MDQ6VXNlcjIwMzA2MDU=',
  'avatar_url': 'https://avatars1.githubusercontent.com/u/2030605?v=4',
  'gravatar_id': '',
  'url': 'https://api.github.com/users/area73',
  'html_url': 'https://github.com/area73',
  'followers_url': 'https://api.github.com/users/area73/followers',
  'following_url': 'https://api.github.com/users/area73/following{/other_user}',
  'gists_url': 'https://api.github.com/users/area73/gists{/gist_id}',
  'starred_url': 'https://api.github.com/users/area73/starred{/owner}{/repo}',
  'subscriptions_url': 'https://api.github.com/users/area73/subscriptions',
  'organizations_url': 'https://api.github.com/users/area73/orgs',
  'repos_url': 'https://api.github.com/users/area73/repos',
  'events_url': 'https://api.github.com/users/area73/events{/privacy}',
  'received_events_url': 'https://api.github.com/users/area73/received_events',
  'type': 'User',
  'site_admin': false,
  'name': 'Rodrigo',
  'company': 'Area 73',
  'blog': 'http://www.siete3.com',
  'location': 'Madrid',
  'email': null,
  'hireable': true,
  'bio': null,
  'public_repos': 102,
  'public_gists': 21,
  'followers': 7,
  'following': 59,
  'created_at': '2012-07-24T00:25:37Z',
  'updated_at': '2019-03-07T11:09:21Z',
};

describe('Card', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Card key={props.id}{...props}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
