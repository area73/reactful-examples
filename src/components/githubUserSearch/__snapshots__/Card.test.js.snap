// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="card"
>
  <img
    className="card_thumb"
    src="https://avatars1.githubusercontent.com/u/2030605?v=4"
  />
  <div
    className="card__info"
  >
    <div
      className="card__substract"
    >
      <a
        href=""
        onClick={[Function]}
      >
        X
      </a>
    </div>
    <div
      className="card__name"
    >
      Rodrigo
    </div>
    <div
      className="card__company"
    >
      Area 73
    </div>
  </div>
</div>
`;
